android/2_build_toolchain.sh

```diff
@@ -66,24 +78,33 @@ function install_lib {
 	cd ..
 }
 
-# Install mpg123
-function install_lib_mpg123() {
-	cd mpg123-1.23.6
-	CPPFLAGS="$CPPFLAGS -DHAVE_MMAP" ./configure --host=$TARGET_HOST --prefix=$PLATFORM_PREFIX \
-		--disable-shared --enable-static \
-		--enable-fifo=no --enable-ipv6=no --enable-network=no --enable-int-quality=no \
-		--with-cpu=generic --with-default-audio=dummy
+# generic cmake library installer
+function install_lib_cmake {
+	cd $1
+	shift
+	rm -rf CMakeCache.txt CMakeFiles/
+	cmake . -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+		-DBUILD_SHARED_LIBS=OFF -DCMAKE_INSTALL_PREFIX=$PLATFORM_PREFIX $@
 	make clean
 	make -j$NBPROC
 	make install
 	cd ..
 }
 
+# Install mpg123
+function install_lib_mpg123 {
+	export CPPFLAGSOLD=$CPPFLAGS
+	export CPPFLAGS="$CPPFLAGS -DHAVE_MMAP"
+	install_lib mpg123-1.25.6 --enable-fifo=no --enable-ipv6=no --enable-network=no \
+		--enable-int-quality=no --with-cpu=generic --with-default-audio=dummy
+	export CPPFLAGS=$CPPFLAGSOLD
+}
+
 # Install SDL2
 function install_lib_sdl {
 	# $1 => platform (armeabi armeabi-v7a x86 mips)
 
-	cd SDL2-2.0.4
+	cd SDL2-2.0.6
 	echo "APP_STL := gnustl_static" > "jni/Application.mk"
```

Comment:
I guess this should be changed to `c++_static`.

Reply:
Good find. Hmm, guess the only reason this didn't crash is the lack of c++ in SDL

Reply:
yep.

Reply:
Yepp, should default to clang by now.